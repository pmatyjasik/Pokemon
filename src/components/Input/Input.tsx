"use client";

import { TextField, TextFieldProps } from "@mui/material";
import styled from "@emotion/styled";

const StyledTextField = styled(TextField)(({ theme }) => ({
    "& .MuiOutlinedInput-root": {
        fontSize: 14,
        borderRadius: theme.shape.borderRadius,
        height: theme.spacing(12),
        "& fieldset": {
            borderColor: theme.palette.grey[400],
            borderWidth: 1,
        },
        "&:hover fieldset": {
            borderColor: theme.palette.primary.main,
        },
        "&.Mui-focused fieldset": {
            boxShadow: theme.shadows[1],
            borderWidth: 1,
        },
    },
    "& .MuiInputBase-input": {
        padding: "14px 10px",
        color: theme.palette.grey[100],
    },
    "& .MuiInputBase-input::placeholder": {
        fontSize: 14,
        color: theme.palette.grey[200],
        opacity: 1,
    },
}));

export type InputProps = TextFieldProps;

export const Input = (props: InputProps) => <StyledTextField {...props} />;
