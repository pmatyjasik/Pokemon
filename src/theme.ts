"use client";

import { createTheme, Shadows } from "@mui/material/styles";
import { ThemeOptions } from "@mui/material/styles";
import { NextFont } from "next/dist/compiled/@next/font";
import localFont from "next/font/local";

const IbmVga: NextFont = localFont({
    src: "./fonts/IBM_VGA.woff",
    display: "swap",
    style: "normal",
    weight: "400",
});

const themeOptions: ThemeOptions = {
    palette: {
        mode: "light",
        background: {
            default: "rgba(255, 255, 255, 1)",
        },
        primary: {
            main: "rgba(151, 71, 255, 1)",
            dark: "rgba(113, 53, 191, 1)",
            light: "rgba(151, 71, 255, 0.25)",
        },
        grey: {
            100: "rgba(42, 42, 42, 1)",
            200: "rgba(127, 127, 127, 1)",
            300: "rgba(228, 228, 228, 1)",
            400: "rgba(238, 238, 238, 1)",
        },
        error: {
            main: "rgba(255, 78, 78, 1)",
        },
        action: {
            disabled: "rgba(0, 0, 0, 0.2)",
        },
    },
    spacing: 4,
    typography: {
        fontFamily: IbmVga.style.fontFamily,
        fontWeightMedium: 400,
        fontSize: 16,
        h6: {
            fontSize: "12px",
        },
        caption: {
            fontSize: "10px",
        },
    },
    shape: {
        borderRadius: 2,
    },
    shadows: [
        "none",
        "0px 0px 0px 4px rgba(151, 71, 255, 0.25)",
        "0px 4px 10px 2px rgba(0, 0, 0, 0.1)",
        ...Array(20).fill("none"),
    ] as Shadows,
};

export const theme = createTheme(themeOptions);
