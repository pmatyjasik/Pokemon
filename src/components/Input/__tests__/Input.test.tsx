import { render, screen } from "@testing-library/react";
import { Input, InputProps } from "@/components/Input/Input";
import { theme } from "@/theme";
import { ThemeProvider } from "@mui/material";

const renderComponent = (props: InputProps) => {
    return render(
        <ThemeProvider theme={theme}>
            <Input {...props} />
        </ThemeProvider>
    );
};

const testValue = "Test Input";

describe("Input Component", () => {
    it("Matches DOM Snapshot", () => {
        const { asFragment } = renderComponent({ placeholder: testValue });

        expect(asFragment()).toMatchSnapshot();
    });

    test("renders the Input component", () => {
        renderComponent({ placeholder: testValue });

        const inputElement = screen.getByPlaceholderText(testValue);

        expect(inputElement).toBeInTheDocument();
    });
});
