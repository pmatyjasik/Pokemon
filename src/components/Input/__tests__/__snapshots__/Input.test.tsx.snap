// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component Matches DOM Snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root css-e5ab0l-MuiFormControl-root-MuiTextField-root"
  >
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-pk2gai-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        id=":r0:"
        placeholder="Test Input"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-ihdtdm"
        >
          <span
            class="notranslate"
          >
            â€‹
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
